Lib/http/server.py

```diff
@@ -902,6 +942,30 @@ def guess_type(self, path):
             return guess
         return 'application/octet-stream'
 
+    def parse_range(self):
+        """Return a tuple of (start, end) representing the range header in
+        the HTTP request. If the range header is missing or not resolvable,
+        None is returned. This only supports single part ranges.
+
+        """
+        range_header = self.headers.get('range')
+        if not range_header:
+            return None
+        m = RANGE_REGEX_PATTERN.match(range_header)
```

Comment:
If the regex pattern is only used for `.match`, we could declare

```py
_range_regex_match = re.compile(...).match
```

instead.

Reply:
https://github.com/python/cpython/pull/118949/files/1784faab3a4c12aa898870674b9236cf03c724c3#diff-10e6db6c775e351ba2b55c4ad5e177191afe25d4942d1d2162d6e10505788a81R129

I tried to keep the naming convention consistent with the existing `DEFAULT_ERROR_CONTENT_TYPE` variable, which is just one line above `RANGE_REGEX_PATTERN`.

Should I change both variables or just leave them as they are? (Personally I don't think there's a big difference here)

Reply:
Up to you. But as I did my review at 1 AM, I now remember that micro optimization of attributes lookup is now meaningless and may hinder future JIT optimizations. So let's keep it as it's currently is.

Reply:
Up to you. But as I did my review at 1 AM, I now remember that micro optimization of attributes lookup is now meaningless and may hinder future JIT optimizations. So let's keep it as it's currently is.